src/runtime/auto-prompt-manager.ts

```diff
@@ -294,20 +309,42 @@ export class AutoPromptManager {
       params.autoPromptType
     )!;
 
-    // Default isClosable to what is set in the page config.
-    // Otherwise, the prompt is blocking for publications with a
-    // subscription revenue model, while all others can be dismissed.
-    this.isClosable_ = params.isClosable ?? !this.isSubscription_();
+    let potentialAction;
+    if (this.actionOrchestrationExperiment_ && !!article.actionOrchestration) {
+      // FPA M0.5 Flow: get next Intervention of the Targeted Funnel.
+      const nextIntervention = await this.getTargetedInterventionOrchestration(
+        clientConfig,
+        article,
+        params.contentType
+      );
+
+      if (!!nextIntervention) {
+        // Fallback to content type for unspecified?
```

Comment:
Sounds like a good idea. What do you think about organizing the logic using switch-case?